test/lib/waitUntilCalled.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * A function that wraps a sinon stubbed function and returns a Promise
+ * when this stub was called.
+ *
+ * The stub that has been passed in will be setup to call the wrapped function
+ * directly, then trigger the returned Promise to resolve.
+ *
+ * @param {Object} stub - A sinon stub of a function
+ * @param {unknown} [wrappedThis] - The object the stubbed function was called on, if any (i.e. the `this` value)
+ * @returns {Promise} A Promise that resolves when the stub has been called
+ */
+export default function waitUntilCalled (stub, wrappedThis = null) {
+  let wasCalled
+  const stubHasBeenCalled = new Promise((resolve) => {
+    wasCalled = resolve
+  })
+  stub.callsFake((...args) => {
+    if (stub.wrappedMethod) {
+      stub.wrappedMethod.call(wrappedThis, ...args)
+    }
+    wasCalled()
+  })
```

Comment:
So, does this function assume that the stub hasn't been modified with e.g. `callsFake` before? We should note that in the docstring if so.

Reply:
Excellent point! Yes, I think it does assume that. I will update the description.

Reply:
I've tested this briefly, and it looks like it does still work if the stub has pre-existing behavior. But it will overwrite it. I'll add a warning explaining that it will overwrite previous fake behavior.

Reply:
Addressed here: https://github.com/MetaMask/metamask-extension/pull/9755/commits/d731f74f7ea6383cd9dd36834fad4cae3d530cfa